src/lib/autodiscovery/autodiscovery.py

```diff
@@ -0,0 +1,110 @@
+    ]
+    discovered_metrics_descriptors = list(
+        filter(
+            lambda descriptor: descriptor.gcpOptions.valueType.upper() != "STRING",
+            discovered_metrics_descriptors,
+        )
+    )
+    discovered_metrics_descriptors = list(
+        filter(
+            lambda descriptor: discovered_resource_type in descriptor.monitored_resources_types,
+            discovered_metrics_descriptors,
+        )
+    )
+    return discovered_metrics_descriptors
+
+
+async def enrich_services_autodiscovery(
+    gcp_services_list: List[GCPService], gcp_session: ClientSession, token: str
+):
+    start_time = time.time()
+    logging_context.log("Adding metrics using autodiscovery")
+
+    bucket_gcp_services = list(
+        filter(lambda x: discovered_resource_type in x.name, gcp_services_list)
+    )
+
+    existing_metric_list = []
+    for service in bucket_gcp_services:
+        existing_metric_list.extend(service.metrics)
+
+    discovered_metric_descriptors = await get_metric_descriptors(gcp_session, token)
+
+    existing_metric_names = [
+        existing_metric.google_metric for existing_metric in existing_metric_list
+    ]
+    missing_metrics_list = []
+
+    for descriptor in discovered_metric_descriptors:
+        if descriptor.value not in existing_metric_names:
+            missing_metrics_list.append(Metric(**asdict(descriptor)))
+    logging_context.log(f"In Extension we have this amount of metrics: {len(existing_metric_list)}")
+    logging_context.log(
+        f"Resource type: {discovered_resource_type} have this amount of metrics: {len(discovered_metric_descriptors)}"
+    )
+    logging_context.log(f"Adding {len(missing_metrics_list)} metrics")
+    logging_context.log(
+        f"Adding metrics: {[metric.google_metric for metric in missing_metrics_list]}"
+    )
+
+    for service in gcp_services_list:
+        if service.name == discovered_resource_type and service.feature_set == "default_metrics":
+            service.metrics.extend([metric for metric in missing_metrics_list])
+
+    end_time = time.time()
+
+    logging_context.log(f"Elapsed time in autodiscovery: {end_time-start_time} s")
+    return gcp_services_list
+
+
+async def prepare_services_autodiscovery_polling(
+    current_services: List[GCPService],
+) -> List[GCPService]:
+    try:
+        async with init_gcp_client_session() as gcp_session:
+            token = await create_token(logging_context, gcp_session)
+            if not token:
+                raise Exception("Failed to fetch token")
+
+            autodiscovery_fetch_result = await enrich_services_autodiscovery(
+                current_services, gcp_session, token
+            )
+
+            return autodiscovery_fetch_result
+    except Exception as e:
+        logging_context.error(
+            f"Failed to prepare autodiscovery new services metrics, will reuse from configuration file; {str(e)}"
```

Comment:
What do you mean "will reuse from configuration file"? 
Do you mean "will not enrich with autodiscovery"?

Reply:
> "will not enrich with autodiscovery"
Exactly. You suggest a different log ?